### PR TITLE
add network policy template

### PR DESCRIPTION
**Pull request by OsamaKhan220**

### CODE REVIEW

**Comment by stakater-user** [comment]
@OsamaKhan220 Validation Successful

File: application/templates/networkpolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+apiVersion: "networking.k8s.io/v1"
+kind: NetworkPolicy
+metadata:
+  labels:
+{{ include "application.labels.stakater" . | indent 4 }}
+{{ include "application.labels.chart" . | indent 4 }}
+{{- if .Values.networkPolicy.additionalLabels }}
+{{ toYaml .Values.networkPolicy.additionalLabels | indent 4 }}
+{{- end }}
+{{- if or .Values.networkPolicy.annotations }}
```
Comment (hanzala1234) [inline application/templates/networkpolicy.yaml:12]: shouldn't have or condition

File: application/templates/networkpolicy.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  annotations:
+{{ toYaml .Values.networkPolicy.annotations | indent 4 }}
+{{- end }}
+  name: {{ template "application.name" . }}-network-policy
```
Comment (hanzala1234) [inline application/templates/networkpolicy.yaml:16]: we are not following a naming convention where we mention resource type in its name.  you can remove -network-policy

File: application/templates/networkpolicy.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  namespace: {{ template "application.namespace" . }}
+spec:
+  podSelector:
+    matchLabels:
+      app: {{ template "application.name" . }}
```
Comment (hanzala1234) [inline application/templates/networkpolicy.yaml:21]: we should add multiple selectors to make sure its applied to one pod. check this for example
https://github.com/stakater-charts/application/blob/master/application/templates/service.yaml#L19

File: application/templates/networkpolicy.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+{{- end }}
+{{- if .Values.networkPolicy.egress }}
+  egress:
+{{ toYaml .Values.networkPolicy.egress | indent 4 }}
+{{- end }}
+{{- if .Values.networkPolicy.ingress }}
+  ingress:
+{{ toYaml .Values.networkPolicy.ingress | indent 4 }}
+{{- end }}
+
```
Comment (hanzala1234) [inline application/templates/networkpolicy.yaml:39]: Remove this extra line

File: application/values-test.yaml [hunk 5]
```diff
@@ -628,6 +628,45 @@ externalSecret:
...
+  ingress:
+  - from:
+    - ipBlock:
+        cidr: 172.17.0.0/16
+        except:
+        - 172.17.1.0/24
+    - namespaceSelector:
+        matchLabels:
+          project: myproject
+    - podSelector:
```
Comment (hanzala1234) [inline application/values-test.yaml:654]: I dont think so we should be adding it over here. it should have been picked by application itself

File: application/values-test.yaml [hunk 6]
```diff
@@ -1139,17 +1178,3 @@ grafanaDashboard:
           "version": 3
         }
 
-cronJob:
```
Comment (hanzala1234) [inline application/values-test.yaml:1142]: why removing this?

File: application/values.yaml [hunk 7]
```diff
@@ -695,6 +695,45 @@ externalSecret:
...
+  ingress:
+  - from:
+    - ipBlock:
+        cidr: 172.17.0.0/16
+        except:
+        - 172.17.1.0/24
+    - namespaceSelector:
+        matchLabels:
+          project: myproject
+    - podSelector:
```
Comment (hanzala1234) [inline application/values.yaml:721]: should have been picked itslef

**Comment by hanzala1234** [comment]
please add add README

**Comment by rasheedamir** [comment]
@OsamaKhan220 resolve the merge conflicts

**Review by hanzala1234 — APPROVED** [review]

**Pull request merged**